### PR TITLE
move from custom virtualenv version to `venv` module

### PR DESCRIPTION
**Pull request by andre-merzky**
We have relied on a specific `virtualenv` version for pilot bootstrapping for a long time.  That worked well as it gave us a stable virtualenv stack to work with.  Now that version has fallen too far behind stock python (see #2845) and it's time to revisit that approach.  This PR switches back to the default `venv` module which has become quite stable over the last releases.  This PR though should wait for the matrix tests to kick in (see #2846) before being considered for merge.

### CODE REVIEW

**Review by mtitov — COMMENTED** [review]
This update is very welcome, thank you, Andre!
Changes looks good, just need to cleanup some leftovers - every [reference of `virtenv_dist`](https://github.com/radical-cybertools/radical.pilot/search?q=virtenv_dist), we wouldn't need this anymore - in `bootstrap_0.sh`, `pmgr/launching/base.py` and resource configs

**Comment by andre-merzky** [comment]
> every [reference of `virtenv_dist`](https://github.com/radical-cybertools/radical.pilot/search?q=virtenv_dist)

Ah, of course!  Thanks, that's cleaned up now.

File: src/radical/pilot/agent/bootstrap_0.sh [hunk 1]
```diff
@@ -1445,7 +1442,6 @@ while getopts "a:b:cd:e:f:g:h:i:j:m:p:r:s:t:v:w:x:y:z:" OPTION; do
         d)  SDISTS="$OPTARG"                  ;;
         e)  pre_bootstrap_0 "$OPTARG"         ;;
         f)  FORWARD_TUNNEL_ENDPOINT="$OPTARG" ;;
-        g)  VIRTENV_DIST="$OPTARG"            ;;
```
Comment (mtitov) [inline src/radical/pilot/agent/bootstrap_0.sh:1504]: really the last one thing - L1440 option character `g` and its description in L1420

**Review by mtitov — APPROVED** [review]
LGTM! just one last cleanup is an option description in bootstrapper

**Comment by andre-merzky** [comment]
> LGTM! just one last cleanup is an option description in bootstrapper

Man, am I glad that you review this! :-)  Fixed!

File: src/radical/pilot/agent/bootstrap_0.sh (hunk 1 above)
Comment (mtitov) [reply src/radical/pilot/agent/bootstrap_0.sh:1504]: thanks! :)

**Comment by andre-merzky** [comment]
Alas, I encountered a number of Python deployments which don't have the `venv` installed by default.  ARGH.  So we have to revert this before release, or at least have the old `virtenv` fetching mechanism as fallback...

**Comment by mtitov** [comment]
@andre-merzky, I don't understand that, as I see it - `venv` is part of python since 3.5 ([link](https://docs.python.org/3.6/library/venv.html)), can you please expand on errors you've encountered?

**Comment by mtitov** [comment]
> The CPython distribution allows you to disable individual built-in packages. For some reason, Linux distros frequently do this, and, e.g. on Ubuntu, you have to explicitly install the `python3-venv` package, or the system python won't have the `venv` module. I don't remember seeing such problems with lmod / environment modules, but it is a common problem with the system python installation.

@eirrgang ah, I see, thank you Eric!

**Pull request merged**